### PR TITLE
testing/knot-resolver4: fix sigsegv in stats module init

### PR DESCRIPTION
**Pull request by tcely**

### CODE REVIEW

**Comment by maxice8** [comment]
i think pkgrel needs to be bumped

**Comment by maxice8** [comment]
squash all into a single commit

File: testing/knot-resolver4/stats-module-memset-upstreams.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--- a/modules/stats/stats.c
++++ b/modules/stats/stats.c
+@@ -498,6 +498,7 @@
+ 	}
+ 	for (size_t i = 0; i < UPSTREAMS_COUNT; ++i) {
+ 		struct sockaddr *sa = (struct sockaddr *)&data->upstreams.q.at[i];
++		memset(sa, 0, sizeof(struct sockaddr));
```
Comment (Ikke) [inline testing/knot-resolver4/stats-module-memset-upstreams.patch]: I'm by no means an expert, but why would it be necessary / make sense to zero-ize `sa` after assigning it a value? Could you not just ignore the assignment and immediately set sa to 0?

Comment (tcely) [reply testing/knot-resolver4/stats-module-memset-upstreams.patch]: Assigning the sa pointer to poiont at the item in the `q` array can not be skipped. We want to zero-ize the `sockaddr` struct in the array before we attempt to set a field in that struct.

Comment (Ikke) [reply testing/knot-resolver4/stats-module-memset-upstreams.patch]: Ok, thanks for the explanation.

Comment (tcely) [reply testing/knot-resolver4/stats-module-memset-upstreams.patch]: You're welcome. As to why this seems to be needed for us and not others, I am not clear on that.

**Pull request closed without merging**